src/utils/url.js

```diff
@@ -83,22 +83,23 @@ export const getEmojiUrl = (unicode: string): string =>
   `/static/generated/emoji/images/emoji/unicode/${unicode}.png`;
 
 export const getNarrowFromLink = (url: string, realm: string, users: User[]): Narrow => {
-  const paths = getPathsFromUrl(url, realm);
-
-  if (isGroupLink(url, realm)) {
-    const recipients = paths[1].split('-')[0].split(',');
-    return groupNarrow(
-      recipients.map((recipient: string) => getUserById(users, parseInt(recipient, 10)).email),
-    );
-  } else if (isTopicLink(url, realm)) {
-    return topicNarrow(
-      decodeURIComponent(transformToEncodedURI(paths[1])),
-      decodeURIComponent(transformToEncodedURI(paths[3])),
-    );
-  } else if (isStreamLink(url, realm)) {
-    return streamNarrow(decodeURIComponent(transformToEncodedURI(paths[1])));
-  } else if (isSpecialLink(url, realm)) {
-    return specialNarrow(paths[1]);
+  if (isUrlInAppLink(url, realm)) {
```

Comment:
What happens if `isUrlInAppLink` returns `false`?
We will not return anything, and that looks like a bug.

You should prefer an 'early return`. 
So something like:
```js
if (!isUrlInAppLink(url, realm)) {
  return ...;
}
```

That makes the code simpler and the diff smaller.

Reply:
It returns `HOME_NARROW` if `isUrlInAppLink` is false.

Changed this to have a early return as you suggested.